java-checks/src/test/files/checks/CORSCheck.java

```diff
@@ -133,20 +136,40 @@ public CorsFilter corsFilter4() {
       public CorsFilter corsFilter4() {
         UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
         CorsConfiguration config = new CorsConfiguration();
-        config.addAllowedOrigin("*"); // Noncompliant [[secondary=137,138]]
+        config.addAllowedOrigin("*"); // Noncompliant [[secondary=140,141]]
         config.applyPermitDefaultValues();
         config.applyPermitDefaultValues();
-        config.addAllowedOrigin("*"); // Noncompliant [[secondary=137,138]]
+        config.addAllowedOrigin("*"); // Noncompliant [[secondary=140,141]]
         return new CorsFilter(source);
       }
     }
     UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
     CorsConfiguration config = new CorsConfiguration();
-    config.addAllowedOrigin("*"); // Noncompliant [[secondary=146,147]]
+    config.addAllowedOrigin("*"); // Noncompliant [[secondary=149,150]]
     config.applyPermitDefaultValues();
     config.applyPermitDefaultValues();
-    config.addAllowedOrigin("*"); // Noncompliant [[secondary=146,147]]
+    config.addAllowedOrigin("*"); // Noncompliant [[secondary=149,150]]
     return new CorsFilter(source);
   }
 
+  class S5122_Insecure implements WebMvcConfigurer {
```

Comment:
I don't see the interest of such complicated methods to tests this. Keep the test case minimal to what we really tests.  For instance, invocations of `allowedMethods`, `allowCredentials`, etc. are not related to the behavior of the rule.